packages/zwave-js/src/lib/serialapi/capability/LongRangeSetupMessages.ts

```diff
@@ -0,0 +1,169 @@
+@expectedResponse(FunctionType.GetLongRangeChannel)
+@priority(MessagePriority.Controller)
+export class GetLongRangeChannelRequest extends Message {}
+
+export interface LongRangeChannelResponseOptions extends MessageBaseOptions {
+	longRangeChannel: LongRangeChannel;
+}
+
+export class LongRangeChannelMessageBase extends Message {
+	public constructor(
+		host: ZWaveHost,
+		options:
+			| MessageDeserializationOptions
+			| LongRangeChannelResponseOptions,
+	) {
+		super(host, options);
+
+		if (gotDeserializationOptions(options)) {
+			// BUGBUG: verify length?
+			switch (this.payload[0]) {
+				case 0x01:
+					this.longRangeChannel = LongRangeChannel.A;
+					break;
+				case 0x02:
+					this.longRangeChannel = LongRangeChannel.B;
+					break;
+				default:
+					this.longRangeChannel = LongRangeChannel.Unknown;
+					break;
+			}
+		} else {
+			this.longRangeChannel = options.longRangeChannel;
+		}
+	}
+
+	public longRangeChannel: LongRangeChannel;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(1);
+
+		switch (this.longRangeChannel) {
+			default: // Don't use reserved values, default back to A
+			case LongRangeChannel.A:
+				this.payload[0] = LongRangeChannel.A;
+				break;
+
+			case LongRangeChannel.B:
+				this.payload[0] = LongRangeChannel.B;
+		}
+
+		return super.serialize();
+	}
+}
+
+@messageTypes(MessageType.Response, FunctionType.GetLongRangeChannel)
+@priority(MessagePriority.Controller)
+export class GetLongRangeChannelResponse extends LongRangeChannelMessageBase {}
+
+@messageTypes(MessageType.Request, FunctionType.SetLongRangeChannel)
+@expectedResponse(FunctionType.SetLongRangeChannel)
+@priority(MessagePriority.Controller)
+export class SetLongRangeChannelRequest extends LongRangeChannelMessageBase {
+}
+
+export interface SetLongRangeChannelResponseOptions extends MessageBaseOptions {
+	responseStatus: number;
+}
+
+// BUGBUG:  move someplace common, see Spec 4.2.5
+//
+// Although, that section refers to a "callback message", and SetZWaveLongRangeChannel doesn't define a callback message...
+export enum ResponseStatus {
+	FAILED = 0x00,
+}
+
+export class ResponseStatusMessageBase extends Message
+	implements SuccessIndicator
+{
+	public constructor(
+		host: ZWaveHost,
+		options: MessageDeserializationOptions,
+	) {
+		super(host, options);
+		this._status = this.payload[0];
+	}
+
+	private _status: ResponseStatus;
+	public get getStatus(): ResponseStatus {
+		return this._status;
+	}
+
+	public isOK(): boolean {
+		return this._status != ResponseStatus.FAILED;
+	}
+}
+
+@messageTypes(MessageType.Response, FunctionType.SetLongRangeChannel)
+export class SetLongRangeChannelResponse extends ResponseStatusMessageBase {}
+
+export interface LongRangeShadowNodeIDsRequestOptions
+	extends MessageBaseOptions
+{
+	shadowNodeIds: number[];
+}
+const LONG_RANGE_SHADOW_NODE_IDS_START = 2002;
+const NUM_LONG_RANGE_SHADOW_NODE_IDS = 4;
```

Comment:
Probably should be in a header somewhere.

Also, I didn't provide an API to set this stuff. Presumably these are for when you want your LR-controller to show up with a LR nodeId? I don't know.

Reply:
same as the others I suppose -> `packages/core/src/consts/index.ts`

Reply:
IMO if we have these messages, we should expose an API on the controller for them.